app/presenters/ht_registration_presenter.rb

```diff
@@ -105,7 +124,29 @@ def show_detail_reverse_lookup
   end
 
   def show_detail_scoped_affiliation
-    env["HTTP_X_SHIB_EDUPERSONSCOPEDAFFILIATION"]
+    affiliation = env["HTTP_X_SHIB_EDUPERSONSCOPEDAFFILIATION"] || ""
+    # Check if scoped affiliation is faculty@, staff@, member@, or employee@
+    # AND matches the allowed affiliations for the IdP the user logged in with
+    badge = ""
+    unless Otis.config.registration.auth_exceptions.member? inst_id
+      badge_id = scoped_affiliation_valid? && scoped_affiliation_match? ? :ok : :mismatch
+      badge = BADGES[badge_id].label_span
+    end
+    affiliation + " " + badge
+  end
+
+  def scoped_affiliation_valid?
+    return false unless env["HTTP_X_SHIB_EDUPERSONSCOPEDAFFILIATION"].present?
+
+    affiliations = env["HTTP_X_SHIB_EDUPERSONSCOPEDAFFILIATION"].downcase.split(";")
+    affiliations.any? { |affiliation| self.class::VALID_AFFILIATIONS.match? affiliation }
+  end
+
+  def scoped_affiliation_match?
+    return false unless env["HTTP_X_SHIB_EDUPERSONSCOPEDAFFILIATION"].present?
+
+    affiliations = env["HTTP_X_SHIB_EDUPERSONSCOPEDAFFILIATION"].downcase.split(";")
+    affiliations.any? { |affiliation| ht_institution.allowed_affiliations.downcase.include? affiliation }
```

Comment:
I suspect this is why we're seeing a mismatch in testing -- `ht_institution.allowed_affiliations` is a regex

Reply:
Curses, those pesky DB regexes.